rfcs/NNNN-decorator-roadmap.md

```diff
@@ -0,0 +1,217 @@
+
+### Standard Decorators
+
+Standard decorators have a much different design than TypeScript's experimental decorators. Rather than getting access to the class and making dynamic, imperative changes to the class, standard decorators have no direct access to the class or the class member they're decorating and must instead return an object describing a replacement for the class member. For instance class accessor decorators return an object with get and set methods.
+
+This makes standard decorator usage must more static than experimental decorators. They can't add new class members, and can't change the type of member they decorate, and ReactiveElement's `static createProperty()` API is simply not callable.
+
+The good news is that our decorator implementations are likely to be smaller and simpler. For example, the two main functions of our `@property()` decorator are to replace a field with a getter/setter pair that calls `requestUpdate()` in the setter, and to store metadata about the field for use in attribute reflection.
+
+For example, a simplified decorator that only replaces an accessor might look like:
+
+```ts
+export const property =
+  (options?: PropertyDeclaration) =>
+  <C extends ReactiveElement, V>(
+    _target: ClassAccessorDecoratorTarget<C, V>,
+    context: ClassAccessorDecoratorContext<C, V>
+  ): ClassAccessorDecoratorResult<C, V> => {
+    const {access, name} = context;
+    return {
+      get(this: C) {
+        return access.get(this);
+      },
+      set(this: C, v: V) {
+        const oldValue = access.get(this);
+        access.set(this, v);
+        this.requestUpdate(name, oldValue);
+      },
+    };
+  };
+```
+
+This decorator would require the new "auto accessor" feature and `accessor` keyword to use:
+
+```ts
+class MyElement extends LitElement {
+  @property()
+  accessor foo = 42;
+}
+```
+
+## Implementation Considerations
+
+### Implementation Plan
+
+The plan is divided into stages which must be shipped as releases.
+
+#### I. Add standard decorators (non-breaking)
+
+This stage adds support for the new decorators standard in a backward-compatible way for both TypeScript and JavaScript developers.
+
+##### Requirements
+* TypeScript ships decorator metadata
+
+##### Changes
+* Add new module with standard decorators
+    * Place them in a new module: `'lit/std-decorators.js'`
+* Re-export legacy decorators from `'lit/legacy-decorators.js'`
+* Add new module with a new implementation of _experimental_ decorators that work the same way standard decorators do:
+    * A _third_ set of decorators: `'lit/experimental-decorators.js'`
+    * They require `accessor`, replace the accessors on the prototype, do not call `static createProperty()`, store metadata in `[Symbol.metadata]`.
+    * This is so that users stuck on the experimental decorators setting (ie, google3) have a path forward when we remove the infrastructure that supports them.
```

Comment:
It's generally useful for larger projects in general that want to incrementally migrate to the new decorators without multiple tsconfigs with per-file carve outs as they migrate

Reply:
Great note! It's extremely useful to call out that this can be used for incremental migration.

Reply:
added this note